### ISSUE TITLE
Support pthread_setname_np

### ISSUE THREAD

**Issue by ramosian-glider** [google/sanitizers#244]
Originally reported on Google Code with ID 244

```
The interceptor was disabled in asan in:
http://llvm.org/viewvc/llvm-project?view=revision&revision=194711

due to:
 #define COMMON_INTERCEPTOR_SET_THREAD_NAME(ctx, name) SetThreadName(name)
+// Should be asanThreadRegistry().SetThreadNameByUserId(thread, name)
+// But asan does not remember UserId's for threads (pthread_t);
+// and remembers all ever existed threads, so the linear search by UserId
+// can be slow.

We need to fix the 2 issues and reenable the interceptor.
```

Reported by `dvyukov@google.com` on 2013-11-15 07:58:34

**Comment by morehouse** [comment]
@dvyukov:  This has been disabled for almost 5 years now.  Do we still plan to re-enable it?

**Comment by dvyukov** [comment]
I don't know if we plan to re-enable it or not. I guess generally not, but this is a good candidate for the fixit.

**Comment by rdhafidh** [comment]
@dvyukov, please dont, or provide runtime dlopen (if possible) detection whether there is pthread_setname_np or not. I have been satisfied that the chromium code now uses the assumption of the prctl function in linux instead of pthread_setname_np, so that I can even build on old linux kernel 2.6.32 glibc 2.11 server. Thanks

**Comment by dvyukov** [comment]
> @dvyukov, please dont, or provide runtime dlopen (if possible) detection whether there is pthread_setname_np or not.

don't we already provide  runtime detection for all interceptors?

**Comment by rdhafidh** [comment]
Yes, I see but I'm still maintaining my own home grown some build fix against eglibc 2.11 though.

>I don't know if we plan to re-enable it or not. I guess generally not, but this is a good candidate for the fixit.

 I was just worried with your statement for possibilty to drop support in the near future.

**Comment by dvyukov** [comment]
> Yes, I see but I'm still maintaining my own home grown some build fix against eglibc 2.11 though.

Is there any reason for this? What?

> I was just worried with your statement for possibilty to drop support in the near future.

This issue is about _adding_ support, not _dropping_ it.

**Comment by rdhafidh** [comment]
>Is there any reason for this? What?

There are some build fix for compiler-rt to work on debian 6 (squeeze) that I usually build.

>This issue is about adding support, not dropping it.

Then thanks for the clarification.

**Comment by fndry-jmg** [comment]
Is there any progress regarding supporting this? It very very very tedious to have to lookup thread numbers versus names, especially if there is no gdb or this mapping is not logged before a crash. Please please re enable it!
